sycl/test/basic_tests/vectors/ctad_fail.cpp

```diff
@@ -0,0 +1,15 @@
+// RUN: %clangxx -std=c++17 -fsyntax-only -Xclang -verify -Xclang -verify-ignore-unexpected %s
+//==--------------- ctad.cpp - SYCL vector CTAD fail test ------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include <CL/sycl.hpp>
+
+namespace sycl = cl::sycl;
+
+int main() {
+  sycl::vec v(1, .1); // expected-error {{no viable constructor or deduction guide for deduction of template arguments of 'vec'}}
```

Comment:
You could extend your CTAD rule to find the minimal type in which everything is convertible. :-)
In this case, `double`

Reply:
It is easy to do with `std::common_type_t`. I'm weakly in favor of not making the change for the reasons:
- The result of `std::common_type_t` can be unexpected. E.g. `std::common_type_t<uint8_t, uint16_t>` is `int` not `uint16_t` or `unsigned int`. Also the result isn't [associative](https://stackoverflow.com/questions/34414139/is-stdcommon-type-associative).
- I'm not aware of a better solution than `std::common_type_t`.  
- The [deduction guide for `std::array`](https://en.cppreference.com/w/cpp/container/array/deduction_guides) made the same choice 

A counter argument is that `sycl::vec` only allows arithmetic types and for those `std::common_type_t` is the same as `decltype(T() + ...)` and the result is associative.